### ISSUE TITLE
node-sodium build error on Mojave

### ISSUE THREAD

**Issue by ming-codes** [evandcoleman/node-appletv#15]
I see `node-appletv` is on `sodium@2`. I'm having trouble building this on MacOS Mojave, but no problem with `sodium@3`. Perhaps it's time to upgrade?

**Comment by reiterbene** [comment]
I have the same Problem with my Raspberry Pi and `node.js 11.2.0`.
Upgrading to `sodium@3.0.1` also does the trick for me.

**Comment by chimai** [comment]
The same issue, can't install node-appletv on both MacOS and raspberry pi 3.

**Comment by WizBangCrash** [comment]
I installed the following and re-ran the `node-appletv` install and all worked for me:

`sudo apt-get install libtool autoconf automake`

**Comment by twentyNoise** [comment]
> I see `node-appletv` is on `sodium@2`. I'm having trouble building this on MacOS Mojave, but no problem with `sodium@3`. Perhaps it's time to upgrade?

Same here

**Comment by WizBangCrash** [comment]
@twentyNoise If you want to build your own version take a look at the `develop` branch of my fork. This is built with sodium 3.0.2 and mdns 2.4.0. 
All seems to work fine for on macOS 10.14.2 and I've also built it on top of a node 11.6 docker image running on my Synology NAS. I'm doing a bit more testing and then I was going to send @edc1591 a pull request to see if he will update the main repo.
You will need to install the following Homebrew packages to build it on macOS:
`$ brew install libtool autoconf automake`

**Comment by twentyNoise** [comment]
> @twentyNoise If you want to build your own version take a look at the `develop` branch of my fork. This is built with sodium 3.0.2 and mdns 2.4.0.

I can do this specifically for this plugin (node-appletv), but how would I do this for homebridge-theater-mode or homebridge-apple-tv which both install node-appletv themselves?

**Comment by WizBangCrash** [comment]
> > @twentyNoise If you want to build your own version take a look at the `develop` branch of my fork. This is built with sodium 3.0.2 and mdns 2.4.0.
> 
> I can do this specifically for this plugin (node-appletv), but how would I do this for homebridge-theater-mode or homebridge-apple-tv which both install node-appletv themselves?

Good question. Unfortunately I am not familiar with either of those packages so I'm not sure I can help. Sorry.
For my own docker image I added the following statement:
```dockerfile
RUN npm install --unsafe-perm sodium@3.0.2 \
  && npm install git+https://github.com/WizBangCrash/node-appletv.git
```
Are you able to uninstall and then install the packages using the above npm commands?

**Comment by GioCirque** [comment]
I think this is resolved by PR #20

**Comment by twentyNoise** [comment]
> @twentyNoise If you want to build your own version take a look at the `develop` branch of my fork. This is built with sodium 3.0.2 and mdns 2.4.0.
> All seems to work fine for on macOS 10.14.2 and I've also built it on top of a node 11.6 docker image running on my Synology NAS. I'm doing a bit more testing and then I was going to send @edc1591 a pull request to see if he will update the main repo.
> You will need to install the following Homebrew packages to build it on macOS:
> `$ brew install libtool autoconf automake`

@WizBangCrash - I just installed this via your Fork. It installed OK but the key changes that you added don't seem to work. The Tv button, the LongTv button etc. Any idea why?

**Comment by evandcoleman** [comment]
Fixed in 1.0.11
